### ISSUE TITLE
Filter "Hide receivers from the map" and "Hide chase cars from the map" do not work while maps stays open for longer time.

### ISSUE THREAD

**Issue by Torrehoradada** [projecthorus/sondehub-tracker#269]
Hi, this is an issue which may have a reference to issue #267 which is already closed.
If you have one or both of those mentioned filters set, these items are initially not displayed on the map. If you now leave the map open for a while, during the time more and more of the filtered items will reappear. 
![Bildschirmfoto vom 2022-10-19 22-17-13](https://user-images.githubusercontent.com/35318102/196799259-9ddae568-370b-4cac-af12-d3073ef040ca.png)
You can remove those items from the map by setting the filter off and setting the filter on again. Sometimes -I do not know why- you need to repeat this one time to see a change on the map display.
If you now again wait long enough, the filtered items will repeat to reappear.
Greetings!

**Comment by darksidelemm** [comment]
Currently away in a location with poor internet access so cannot verify what's going on here.
Will have a look when I return in a few weeks..

**Comment by Torrehoradada** [comment]
Still the same, the error occures on following filters: 
- Hide receivers from the map
- Hide recovered sondes from the map
- Hide chase cars from the map
Also there is an error with "Only show visible on sidebar" which -when switched on- initially always displays one sonde which is outside of the view area. Moving around that area will eventually remove this one.

**Comment by darksidelemm** [comment]
I can confirm I've seen the behaviour of new receivers showing up after they have been 'hidden'. This is likely due to the websockets handling code not respecting those settings. I'm sure its the case for the other settings too (recovered sondes, chase cars).

I'll have some time to look into this over the next week or so.

**Comment by darksidelemm** [comment]
Hmm, this is an odd one. When those toggles are set, it certainly should clear all existing items off the map, then inhibit any new items from being showed.  Will require more investigation...

**Comment by TheSkorm** [comment]
Is this still an issue?

**Comment by Torrehoradada** [comment]
Of course...  Yes it is!
Seems that Mark did not had the time to work on it after his confirmation...

**Comment by TheSkorm** [comment]
I found a few possible race conditions that should be fixed in https://testing.v2.sondehub.org if you want to give that a go.

**Comment by Torrehoradada** [comment]
Hi, first sorry for the multiple edits....
I checked the 'normal' map and did not see the problem at the first glance, but later some of the chase cars and receivers reappeared...
I had the 'testing' map open for some time and I can confirm that this error seems to have been fixed there.
Comparing the 'testing' and 'normal' map -which now I have been open in parallel at the same area- I do see a bunch of receivers and some chase cars reappearing on the 'normal' map. -"Hide"-filters still set.
All of these do not appear on the 'testing' map. So filtering -at least- has been greatly improved.

**Comment by TheSkorm** [comment]
Thanks for testing!

**Comment by Torrehoradada** [comment]
On the before mentioned 'testing' map I still have the error that often are additional sondes on the list, which are located far outside the visible area. They also may eventually disappear after a while, but that is not for sure.
This is the same error I experienced in my original post and there is no difference between 'testing' and 'normal' map...

**Comment by TheSkorm** [comment]
A new ticket would be best. if you can grab a screenshot or video recording it would be useful. Is it only in test or both?

**Comment by Torrehoradada** [comment]
Its on both

**Comment by Torrehoradada** [comment]
![grafik](https://github.com/user-attachments/assets/c31cd060-4dc0-4346-b9bb-aaf1053d7d53)
Can you see this?

**Comment by Torrehoradada** [comment]
New issue in creation,
This is the one additional sonde with active filter "Only show visible on sidebar". The DFM09 was(still is) far outside the map in Poland... There also are others, -not predictable, sometimes located on other continents...
![Bildschirmfoto vom 2024-11-20 09-09-15](https://github.com/user-attachments/assets/4e8dad78-9cf6-4e51-b87b-940d35e6efbe)

**Comment by TheSkorm** [comment]
testing site should have a fix for the filter issue - https://github.com/projecthorus/sondehub-tracker/pull/370

**Comment by Torrehoradada** [comment]
Wow! You ARE really quick!
Confirm, now no irritating "pops" anymore!
Thank you very much!
